### PR TITLE
[SDK-Types] Sdk types for tx builder

### PR DESCRIPTION
**Pull request by stefan-mysten**
This PR adds the changes needed for the transaction builder PR related to the `sui-sdk-types` crate. In a nutshell:
- adds a basic API for TransactionEffects
- renames InputArgument to Input
- renames the `Unresolved` types in `unresolved` module, and exposes the module itself instead of all the `UnresolvedTypes`.
- adds the needed constructors for the newly renamed `unresolved::Input` type.

### CODE REVIEW

**Comment by stefan-mysten** [comment]
I assume this might break `sui-rest-api` once it's updated to the latest commit/version.

File: crates/sui-sdk-types/src/types/transaction/unresolved.rs [hunk 1]
```diff
@@ -93,31 +95,41 @@ pub struct UnresolvedObjectReference {
...
+/// transaction builder to resolve all unresolved inputs.
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 #[cfg_attr(
     feature = "serde",
     derive(serde_derive::Serialize, serde_derive::Deserialize)
 )]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
-pub struct UnresolvedInputArgument {
+pub struct Input {
+    /// The object kind. This field is only relevant for object inputs.
```
Comment (bmwill) [inline crates/sui-sdk-types/src/types/transaction/unresolved.rs:119]: > This field is only relevant for object inputs.

this is inaccurate, lets remove it

File: crates/sui-sdk-types/src/types/transaction/unresolved.rs [hunk 2]
```diff
@@ -93,31 +95,41 @@ pub struct UnresolvedObjectReference {
...
     #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none",))]
-    pub kind: Option<UnresolvedInputArgumentKind>,
+    pub kind: Option<InputKind>,
+    /// The value of this object. This field is only relevant for Pure kind of inputs.
```
Comment (bmwill) [inline crates/sui-sdk-types/src/types/transaction/unresolved.rs:122]: > This field is only relevant for Pure kind of inputs.

This is probably going to be inaccurate at some point so lets remove it

File: crates/sui-sdk-types/src/types/effects/v2.rs [hunk 3]
```diff
@@ -185,6 +185,23 @@ pub enum IdOperation {
...
+impl TransactionEffectsV2 {
+    /// The status of the execution
+    pub fn status(&self) -> &ExecutionStatus {
+        &self.status
+    }
+
+    /// The epoch when this transaction was executed.
+    pub fn epoch(&self) -> &EpochId {
+        &self.epoch
+    }
```
[the remark this answers is not in the document]
Comment (stefan-mysten) [reply crates/sui-sdk-types/src/types/effects/v2.rs:197]: Fixed

File: crates/sui-sdk-types/src/types/transaction/unresolved.rs [hunk 4]
```diff
@@ -142,17 +157,171 @@ pub struct UnresolvedInputArgument {
...
+    /// Return an owned kind of object with all required fields.
+    pub fn owned(object_id: ObjectId, version: u64, digest: ObjectDigest) -> Self {
+        Self {
+            kind: Some(InputKind::ImmutableOrOwned),
+            object_id: Some(object_id),
+            version: Some(version),
+            digest: Some(digest),
+            ..Default::default()
+        }
+    }
```
[the remark this answers is not in the document]
Comment (stefan-mysten) [reply crates/sui-sdk-types/src/types/transaction/unresolved.rs:178]: Reworked this.

**Comment by stefan-mysten** [comment]
Thanks @bmwill. I think I've addressed the comments.

**Pull request merged**